presto-main/src/main/java/com/facebook/presto/metadata/SignatureBuilder.java

```diff
@@ -63,7 +73,7 @@ public SignatureBuilder typeVariableConstraints(TypeVariableConstraint... typeVa
 
     public SignatureBuilder typeVariableConstraints(List<TypeVariableConstraint> typeVariableConstraints)
     {
-        this.typeVariableConstraints = ImmutableList.copyOf(requireNonNull(typeVariableConstraints, "typeVariableConstraints is null"));
+        this.typeVariableConstraints = unmodifiableList(new ArrayList<>(requireNonNull(typeVariableConstraints, "typeVariableConstraints is null")));
```

Comment:
This is probably no longer needed if you are not moving it to SPI.

Reply:
Good catch